crates/cubecl-hip-sys/build.rs

```diff
@@ -14,15 +76,25 @@ fn main() {
     // default installation path
     paths.push("/opt/rocm".to_string());
 
-    let rocm_path = paths.iter().find(|path| {
-        let hip_path = Path::new(path).join("include/hip");
-        hip_path.exists()
-    });
+    let mut rocm_path_candidates = paths
+        .iter()
+        .filter(|path| {
+            let hip_path = Path::new(path).join("include/hip");
+            hip_path.exists()
+        })
+        .peekable();
+    let have_candidates = rocm_path_candidates.peek().is_some();
+    let rocm_path = rocm_path_candidates.find(|path| check_version(path).unwrap_or_default());
 
     if let Some(valid_rocm_path) = rocm_path {
         println!("cargo:rustc-link-lib=dylib=hiprtc");
         println!("cargo:rustc-link-lib=dylib=amdhip64");
         println!("cargo:rustc-link-search=native={}/lib", valid_rocm_path);
+    } else if have_candidates {
+        panic!(
+            "None of the found ROCm installations match crate version {}",
+            env!("CARGO_PKG_VERSION")
+        );
```

Comment:
If `check_version` returns true, then that path will go into `valid_rocm_path` branch earlier, because `find` function will succeed. This branch is for the situation where there is a path candidate, but it wasn't the right major version.

Example of this compiling in action:

https://github.com/user-attachments/assets/b618e2f1-5cb1-464d-b61b-3be9314e0d41

As you can see in the video, only major version mismatch causes this panic, and minor version only shows a warning